public/projects-list/scripts/glitch.js

```diff
@@ -4,10 +4,102 @@ var $ = require("jquery");
 
 module.exports = {
   init: function() {
-    let banner = $(".glitch-banner");
-    let cta = $(".glitch-cta.underlay");
-    if (banner && cta) {
+    let banner = $(".glitch-banner"),
+      cta = $(".glitch-cta.underlay"),
+      content = $(".content", cta),
+      close = $(".cta-close", content),
+      start = $("button.export-button.start", content),
+      exportPublished = $("button.export-button.published", content),
+      exportUnpublished = $("button.export-button.unpublished", content),
+      csrfToken = $("meta[name='csrf-token']").attr("content"),
+      importURL = $("meta[name='glitch-url']").attr("content"),
+      exportLabel = $("meta[name='export-label']").attr("content"),
+      projectId = undefined,
+      projectUrl = undefined,
+      restoreButtonText = () => {};
+
+    function getToken(url, onSuccess, onError) {
+      fetch(url, {
+        method: "POST",
+        headers: {
+          "X-Csrf-Token": csrfToken
+        }
+      })
+        .then(response => {
+          if (response.status !== 200) {
+            throw new Error("could not retrieve token");
+          }
+          return response.json();
+        })
+        .then(object => o.token)
+        .then(token => onSuccess(token))
+        .catch(e => onError(e));
+    }
+
+    function notifyError(error) {
+      $(".normal", cta).addClass("hidden");
+      $(".error", cta).removeClass("hidden");
+      restoreButtonText();
+    }
+
+    if (banner.length && cta.length) {
       banner.removeClass("hidden");
     }
+
+    // We track the projectId "globally" for a dialog, because there
+    // can only ever be one dialog open at a time, triggered for
+    // one specific project to be exported.
+
+    let runOperation = (evt, button, published) => {
+      let root = published ? `publishedprojects` : `projects`;
+      let url = `/${root}/${projectId}/export/start`;
```

Comment:
you should only use project ID for the projects/ route, and for the publishedprojects/ route you should use the `publishedProjectId`. You'll have access to that on the server so you can render it in a data attribute or just parse it out of the publish_url which you already have access to

Reply:
Wait, why would those two not be linked? We update the content when you "re"publish, so your old page is gone and now it's the new page. Why would that need a separate publication id?

(I'll happily update the code but needing to call the API with a different id seems odd. The server should be able to do the thimbledb.project.id -> publishdb.project.id lookup without the client having to care?)